pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/ProtoWriterToolsTest.java

```diff
@@ -0,0 +1,759 @@
+        bufferedData.flip();
+        assertEquals((definition.number() << TAG_TYPE_BITS) | WIRE_TYPE_DELIMITED.ordinal(), bufferedData.readVarInt(false));
+        int length = bufferedData.readVarInt(false);
+        assertEquals(appleStr, Apple.parseFrom(bufferedData.readBytes(length).toByteArray()).getVariety());
+    }
+
+    @Test
+    void testWriteOneOfMessage() throws IOException {
+        FieldDefinition definition = createOneOfFieldDefinition(MESSAGE);
+        writeMessage(bufferedData, definition, null, (data, out) -> out.writeBytes(data.toByteArray()), Apple::getSerializedSize);
+        bufferedData.flip();
+        assertEquals((definition.number() << TAG_TYPE_BITS) | WIRE_TYPE_DELIMITED.ordinal(), bufferedData.readVarInt(false));
+        int length = bufferedData.readVarInt(false);
+        assertEquals(0, length);
+
+    }
+
+    @Test
+    void testWriteOptionalInteger() {
+        FieldDefinition definition = createOptionalFieldDefinition(INT32);
+        final int valToWrite = randomLargeInt();
+        writeOptionalInteger(bufferedData, definition, valToWrite);
+        bufferedData.flip();
+        assertTypeDelimitedTag(definition);
+        assertEquals(MAX_VAR_INT_SIZE, bufferedData.readVarInt(false));
+        assertVarIntTag(definition.type().optionalFieldDefinition);
+        assertEquals(valToWrite, bufferedData.readVarInt(false));
+    }
+
+    @Test
+    void testWriteOptionalInteger_null() {
+        FieldDefinition definition = createOptionalFieldDefinition(INT32);
+        writeOptionalInteger(bufferedData, definition, null);
+        bufferedData.flip();
+        assertEquals(0, bufferedData.length());
+    }
+
+    @Test
+    void testWriteOptionalLong() {
+        FieldDefinition definition = createOptionalFieldDefinition(INT64);
+        final long valToWrite = randomLargeInt();
+        writeOptionalLong(bufferedData, definition, valToWrite);
+        bufferedData.flip();
+        assertTypeDelimitedTag(definition);
+        assertEquals(MAX_VAR_INT_SIZE, bufferedData.readVarInt(false));
+        assertVarIntTag(definition.type().optionalFieldDefinition);
+        assertEquals(valToWrite, bufferedData.readVarInt(false));
+    }
+
+    @Test
+    void testWriteOptionalLong_null() {
+        FieldDefinition definition = createOptionalFieldDefinition(INT64);
+        writeOptionalLong(bufferedData, definition, null);
+        bufferedData.flip();
+        assertEquals(0, bufferedData.length());
+    }
+
+    @Test
+    void testWriteOptionalFloat() {
+        FieldDefinition definition = createOptionalFieldDefinition(FLOAT);
+        final float valToWrite = randomLargeInt();
+        writeOptionalFloat(bufferedData, definition, valToWrite);
+        bufferedData.flip();
+        assertTypeDelimitedTag(definition);
+        assertEquals(MAX_VAR_FLOAT_SIZE, bufferedData.readVarInt(false));
```

Comment:
`MAX_VAR_FLOAT_SIZE` name is confusing. Floats are not variable, they are fixed 4 bytes. Optional floats are either 0 bytes (null value), or sizeOfDelimited(field, 4) bytes. `OPTIONAL_FLOAT_SIZE` would be a better name, in my opinion. Also, check my comment to `MAX_VAR_FLOAT_SIZE`.

Reply:
I renamed the constants